### PR TITLE
luci-proto-modemmanager: add luci-proto-modemmanager

### PR DESCRIPTION
**Pull request by nickberry17**
Signed-off-by: Nicholas Smith <nicholas.smith@telcoantennas.com.au>

### CODE REVIEW

File: protocols/luci-proto-modemmanager/luasrc/model/network/proto_modemmanager.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+function proto.contains_interface(self, ifc)
+	 return (netmod:ifnameof(ifc) == self:ifname())
+end
+
+netmod:register_pattern_virtual("^mobiledata%-%w")
+
+netmod:register_error_code("CALL_FAILED",	luci.i18n.translate("Call failed"))
+netmod:register_error_code("NO_CID",		luci.i18n.translate("Unable to obtain client ID"))
+netmod:register_error_code("PLMN_FAILED",	luci.i18n.translate("Setting PLMN failed"))
```
Comment (feckert) [inline protocols/luci-proto-modemmanager/luasrc/model/network/proto_modemmanager.lua]: I have not tested it but i think we do not need this file.
All other protocols does not have this file its all handled in JavaScript now.

Comment (jow-) [reply protocols/luci-proto-modemmanager/luasrc/model/network/proto_modemmanager.lua]: The model classes are required yet. Some parts of LuCI still indirectly depend on them.

Comment (feckert) [reply protocols/luci-proto-modemmanager/luasrc/model/network/proto_modemmanager.lua]: Thank you for clarifying. All other protos did not have this.
So I thought this is not necessary.
I haven't worked with the new Javascript client code yet.
Maybe I should do it if I get/have time for this. :-)

**Comment by BKPepe** [comment]
Can this be backported to OpenWrt 19.07? PR to include modemmanager in OpenWrt 19.07 (https://github.com/openwrt/packages/pull/11149) was merged a few months ago.

**Pull request merged**